includes/class-pwa.php

```diff
@@ -135,5 +137,66 @@ public static function caching_strategy_args( $args ) {
 		$args['plugins']['expiration']['maxEntries'] = 50;
 		return $args;
 	}
+
+	/**
+	 * Render Push Notification required HTML files.
+	 */
+	public static function render_push_notification_urls() {
+		$onesignal_cm = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'onesignal' );
+		if ( ! $onesignal_cm->is_configured() ) {
+			return;
+		}
+		if ( isset( $_SERVER['REQUEST_URI'] ) ) { // WPCS: Input var okay.
+			$raw_uri = sanitize_text_field(
+				wp_unslash( $_SERVER['REQUEST_URI'] ) // WPCS: Input var okay.
+			);
+		} else {
+			$raw_uri = '';
+		}
+		$path = null;
+
+		if ( substr( $raw_uri, 0, strlen( '/helper-iframe.html' ) ) === '/helper-iframe.html' ) {
+			$path = dirname( NEWSPACK_PLUGIN_FILE ) . '/assets/raw/amp-web-push-helper-frame.html';
```

Comment:
Interested to know why not use `/assets/raw/amp-web-push-helper-frame.html` directly with the `<amp-web-push>` tag? Instead of going through `/helper-iframe.html` then mapping it.